torchtext/experimental/datasets/sequence_tagging.py

```diff
@@ -0,0 +1,220 @@
+import torch
+
+from torchtext.experimental.datasets import raw
+from torchtext.vocab import build_vocab_from_iterator
+from torchtext.experimental.functional import (
+    vocab_func,
+    totensor,
+    sequential_transforms,
+)
+
+
+def _build_vocab(data, word_transform):
+    total_columns = len(data[0])
+    data_list = [[] for _ in range(total_columns)]
+    vocabs = []
+
+    for line in data:
+        for idx, col in enumerate(line):
+            if idx == 0:
+                col = word_transform(col) if word_transform else col
+                data_list[idx].append(col)
+            else:
+                data_list[idx].append(col)
+
+    for it in data_list:
+        vocabs.append(build_vocab_from_iterator(it))
+
+    return vocabs
+
+
+def _setup_datasets(dataset_name,
+                    train_filename,
+                    valid_filename,
+                    test_filename,
+                    separator,
+                    data_select=("train", "valid", "test"),
+                    root=".data",
+                    vocabs=None,
+                    word_tokenizer=None):
+    train, val, test = DATASETS[dataset_name](train_filename=train_filename,
+                                              valid_filename=valid_filename,
+                                              test_filename=test_filename,
+                                              root=root)
+    raw_data = {
+        "train": [line for line in train] if train else None,
+        "valid": [line for line in val] if val else None,
+        "test": [line for line in test] if test else None
+    }
+
+    word_transform = None
```

Comment:
nit: to keep consistent, I would call `word_transform` as `text_transform`

Reply:
changed